test/e2e/instascale_nodepool_test.go

```diff
@@ -0,0 +1,66 @@
+package e2e
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	. "github.com/project-codeflare/codeflare-common/support"
+	mcadv1beta1 "github.com/project-codeflare/multi-cluster-app-dispatcher/pkg/apis/controller/v1beta1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestInstascaleNodepool(t *testing.T) {
+
+	test := With(t)
+	test.T().Parallel()
+
+	clusterType := GetClusterType(test)
+	if clusterType != HypershiftCluster {
+		test.T().Skipf("Skipping test as not running on an Hypershift cluster, resolved cluster type: %s", clusterType)
+	}
+
+	namespace := test.NewTestNamespace()
+
+	// Test configuration
+	cm := CreateConfigMap(test, namespace.Name, map[string][]byte{
+		// pip requirements
+		"requirements.txt": ReadFile(test, "mnist_pip_requirements.txt"),
+		// MNIST training script
+		"mnist.py": ReadFile(test, "mnist.py"),
+	})
+
+	//create OCM connection
+	connection := CreateOCMConnection(test)
+	defer connection.Close()
+
+	// check existing cluster resources
+	// look for a node pool with a label key equal to aw name - expect NOT to find it
+	test.Expect(GetNodePools(test, connection)).
+		ShouldNot(ContainElement(WithTransform(NodePoolLabels, HaveKey(HavePrefix("test-instascale")))))
```

Comment:
The label here has been updated to `name-namespace` to prevent conflicts.
we could do something similar to: 
```go
// look for a node pool with a label key equal to aw.name-aw.namespace - expect NOT to find it

expectedLabel = fmt.Sprintf("%s-%s", appWrapperName, appWrapperNamespace)

test.Expect(GetNodePools(test, connection)).
	ShouldNot(ContainElement(WithTransform(NodePoolLabels, HaveKey(expectedLabel))))

```

However, this still works, just thought it worth mentioning :)

Reply:
Thanks Eoin! I think using `HavePrefix` is neater but we should use whichever is best practice. What do you think @sutaakar?

Reply:
Updated, thank you both.